### PR TITLE
Move from DockerHub to quay.io

### PR DESCRIPTION
**Pull request by yuvipanda**
Companion PR to
https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3254. See that PR for more information.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ jobs:
         #    https://github.com/jupyterhub/configurable-http-proxy/settings/secrets/actions
         if: startsWith(github.ref, 'refs/tags/')
         run: |
-          docker login -u "${{ secrets.DOCKERHUB_USERNAME }}" -p "${{ secrets.DOCKERHUB_TOKEN }}"
+          docker login -u "${{ secrets.QUAY_USERNAME }}" -p "${{ secrets.QUAY_PASSWORD }}" quay.io
```
Comment (minrk) [inline .github/workflows/publish.yml:71]: This changes the creds, but doesn't change where it's pushed (next step in gettags needs `quay.io` added, I think)

Comment (yuvipanda) [reply .github/workflows/publish.yml:71]: @minrk how about now?

Comment (minrk) [reply .github/workflows/publish.yml:71]: I think that's right, thanks.

**Review by minrk — APPROVED** [review]

**Comment by yuvipanda** [comment]
Thanks @minrk. I'll use workflow dispatch to run this just in case.

**Comment by yuvipanda** [comment]
I just re-pushed 4.6.0 but realized that won't actually trigger the new workflow, only the old one.

I don't want to make a fresh release, so I've just re-run `➜ skopeo copy --multi-arch all docker://jupyterhub/configurable-http-proxy:4.6.0 docker://quay.io/jupyterhub/configurable-http-proxy:4.6.0` and am gonna call it done.

**Pull request merged**